utils/frame/remote-externalities/src/lib.rs

```diff
@@ -351,14 +351,57 @@ where
 		prefix: StorageKey,
 		at: B::Hash,
 	) -> Result<Vec<StorageKey>, &'static str> {
-		let mut last_key: Option<StorageKey> = None;
+		const NUM_PARTS: usize = 4; // number of parts to divide the prefix into
+
+		let mut tasks = Vec::with_capacity(NUM_PARTS);
+
+		// Divide the prefix into N parts
+		let mut start_key = prefix.clone();
+		let mut end_key = prefix.clone();
 		let mut all_keys: Vec<StorageKey> = vec![];
-		let keys = loop {
+
+		for _ in 0..NUM_PARTS {
+			end_key.0[0] = start_key.0[0].wrapping_add(1);
+			let task = self.rpc_get_keys_paged_range(start_key.clone(), end_key.clone(), at);
+			tasks.push(task);
+			start_key.0[0] = end_key.0[0].wrapping_add(1);
+		}
+
+		for task in futures::future::join_all(tasks).await {
+			let mut keys = task?;
+			all_keys.append(&mut keys);
+		}
```

Comment:
I'd prefer if this feature was implemented with a more functional approach rather than procedural, similar to https://github.com/paritytech/substrate/blob/ab8330cd421027d0ebfd613111702fd4fa5e9487/utils/frame/remote-externalities/src/lib.rs#L560-L578

I feel functional is easier to follow and less bug-prone